corehq/apps/sso/tasks.py

```diff
@@ -110,6 +119,64 @@ def send_idp_cert_expires_reminder_emails(num_days):
             )
 
 
+@periodic_task(run_every=crontab(minute=0, hour=2), acks_late=True)
+def auto_deactivate_removed_sso_users():
+    for idp in IdentityProvider.objects.filter(
+        enable_user_deactivation=True,
+        idp_type=IdentityProviderType.ENTRA_ID
+    ).all():
+        idp_users = idp.get_all_members_of_the_idp()
+        usernames_in_account = idp.owner.get_web_users(info_type='username')
+
+        # Get criteria for exempting usernames and email domains from the deactivation list
+        authenticated_domains = AuthenticatedEmailDomain.objects.filter(identity_provider=idp)
+        exempt_usernames = UserExemptFromSingleSignOn.objects.filter(email_domain__in=authenticated_domains
+                                                                     ).values_list('username', flat=True)
+
+        # if the Graph Users API returns an empty list of users we will skip auto deactivation
+        if len(idp_users) == 0:
+            context = get_sso_deactivation_skip_email_context(idp)
+            if not context["to"]:
+                notify_exception(None, f"no admin email addresses for IdP: {idp}")
+            try:
+                for send_to in context["to"]:
+                    send_html_email_async.delay(
+                        context["subject"],
+                        send_to,
+                        context["html"],
+                        text_content=context["plaintext"],
+                        email_from=context["from"],
+                        bcc=context["bcc"],
+                    )
+                    log.info(
+                        "Sent sso user deactivation skipped notification"
+                        "email for %(idp_name)s to %(send_to)s." % {
+                            "idp_name": idp.name,
+                            "send_to": send_to,
+                        }
+                    )
+            except Exception as exc:
```

Comment:
Same comment as above, if there is a specific exception you know could be raised, catch it, otherwise this try catch is likely not needed.